simple_parsing/helpers/serialization/serializable.py

```diff
@@ -734,8 +752,42 @@ def from_dict(
                 )
             continue
 
-        raw_value = obj_dict.pop(name)
-        field_value = decode_field(field, raw_value, containing_dataclass=cls)
+        if field.metadata.get("subgroups", None) and parse_selection:
+            # decode subgroups from dict
+            subgroups_dict = field.metadata.get("subgroups")
+
+            _target = f"__subgroups__@{field.name}"
+            if _target in obj_dict:
+                _target_cls = subgroups_dict[obj_dict[_target]]
+                raw_value = obj_dict.pop(_target)
+
+            if name in obj_dict:
+                raw_value = obj_dict.pop(name)
```

Comment:
Is it desired for this second case to overwrite the first one if `name in obj_dict` ?
```suggestion
            elif name in obj_dict:
                raw_value = obj_dict.pop(name)
```

Reply:
It is necessary because the user would override `{"__subgroups__@[key]": "[selected_key]"` generated by `to_dict` by manually putting `{"[key]": "[selected_key]"}` to `replace`